src/lib/installation/update_repositories_finder.rb

```diff
@@ -265,5 +268,34 @@ def display_fallback_warning
       # display the message in a RichText widget to wrap long lines
       Yast::Report.LongWarning(msg)
     end
+
+    # Runs a block of code handling errors
+    #
+    # If errors should be shown, the helper {catch_registration_errors}
+    # from Registration::ConnectionHelpers will be used.
+    #
+    # Otherwise, errors will be logged and the method will return +false+.
+    #
+    # @params [Boolean] True if errors should be shown to the user. False otherwise.
+    # @return [false, Object] The value returned by the block itself. False
+    #                         if the block failed.
+    #
+    # @see Registration::ConnectionHelpers.catch_registration_errors
+    def handle_registration_errors(show_errors)
+      if show_errors
+        require "registration/connection_helpers"
+        ret = nil
+        success = ::Registration::ConnectionHelpers.catch_registration_errors { ret = yield }
+        success && ret
+      else
+        begin
+          yield
+        rescue StandardError => e
+          log.error("Could not determine update repositories through the registration server: " \
```

Comment:
NP: `log.warn` might be better as we do not report the errors in the automatic mode. They are actually not real errors in this case so treating them as a warning would be more appropriate.